src/main/java/cd/go/authorization/gitlab/client/GitLabClient.java

```diff
@@ -80,10 +83,10 @@ public TokenInfo fetchAccessToken(String code) throws IOException {
 
         final FormBody formBody = new FormBody.Builder()
                 .add("client_id", gitLabConfiguration.applicationId())
-                .add("client_secret", gitLabConfiguration.clientSecret())
```

Comment:
Let's discuss in the linked issue but don't think we want to remove client secrets or use of confidential OAuth2 clients as this appears to do.

https://github.com/gocd-contrib/gitlab-oauth-authorization-plugin/issues/220#issuecomment-2339552804

Reply:
Validated that PKCE works with confidential clients, so have added this back :-)